src/lib/ffi/ffi_rng.cpp

```diff
@@ -57,6 +57,93 @@ int botan_rng_init(botan_rng_t* rng_out, const char* rng_type)
       });
    }
 
+int botan_rng_init_custom(botan_rng_t* rng_out, const char* rng_name, void* context,
+                          int(* get_cb)(uint8_t* out, size_t out_len, void* context),
+                          int(* add_entropy_cb)(const uint8_t input[], size_t length, void* context))
+{
+return ffi_guard_thunk(__func__,[=]() -> int {
+   if(rng_out == nullptr)
+      return BOTAN_FFI_ERROR_NULL_POINTER;
+
+   std::unique_ptr<Botan::RandomNumberGenerator> rng;
```

Comment:
Move this decl to after `Custom_RNG`, then it can just be

`std::unique_ptr<Botan::RandomNumberGenerator> rng(new Custom_RNG(...))`

without the rest